master/internal/core_checkpoint_upstream_intg_test.go

```diff
@@ -0,0 +1,241 @@
+//go:build integration && upstream
+// +build integration,upstream
```

Comment:
Is it possible to just detect when the CI environment variables aren't present, and use t.Skip() or something?

I think all you'd have to change in the circleci config is which circleci security context was given to this test.  I think circleci would automatically not give any security context to non-upstream runs of the test.  There would never be overlapping circleci runs.  Then there wouldn't be changes to master/Makefile either I think.

Reply:
Sounds good. How about skipping it if `CIRCLE_BRANCH` starts with `pull/`? This is the same logic used in `.circleci/config.yaml`